flocker/acceptance/endtoend/test_dockerplugin.py

```diff
@@ -143,9 +148,83 @@ def _create_volume(self, client, name, driver_opts):
             'DriverOpts': driver_opts,
         }
         result = client._result(client._post_json(url, data=data), True)
-        self.addCleanup(client.remove_volume, name)
+        if cleanup:
+            self.addCleanup(client.remove_volume, name)
         return result
 
+    def _test_sized_vol_container(self, cluster, node):
+        """
+        Create a volume with a size, then create a
+        container that can use that volume and run lsbk to test
+        its size. An http server inside the container uses lsblk to
+        return the size of the volume when path is not "/is_http"
+
+        :param cluster: flocker cluster
+        :param node: node the contianer and volume will be on.
+        """
+        client = get_docker_client(cluster, node.public_address)
+        volume_name = random_name(self)
+        # Because flocker/control/schema/types.yml#L250 > 64MiB
+        # "k" and "t" are too big and too small for test
+        # but we to keep some randomness to this test.
+        size_expressions = list(chain.from_iterable((x, x+"b", x+"ib")
+                                for x in ["g"]))
+        # Note: if a user types in 50mib, flocker defaults to
+        # at least 1G because this is AWS minimums etc, so this test
+        # cannot test all expressions but we test g, gib, gb.
+        # Note: openstack cinder must have between 75-100GB :(
+        # http://www.rackspace.com/knowledge_center/product-faq
+        # /cloud-block-storage
+        # https://github.com/ClusterHQ/flocker/blob/
+        # 780b22a443d1f2c10211b12e08d08ab1585099dc/flocker/
+        # node/agents/cinder.py#L471 passes it through
+        size = str(randint(75, 95)) + str(
+            size_expressions[randint(0, len(size_expressions) - 1)])
+        self._create_volume(client, volume_name,
+                            driver_opts={'size': size})
+        http_port = 8080
+        host_port = find_free_port()[1]
+        self.run_python_container(
+            cluster, node.public_address,
+            {"host_config": client.create_host_config(
+                binds=["{}:/sizedvoldata".format(volume_name)],
+                port_bindings={http_port: host_port},
+                restart_policy={"Name": "always"},
+                privileged=True),
+             "ports": [http_port]},
+            SCRIPTS.child(b"lsblkhttp.py"),
+            [u""], client=client)
+
+        d = assert_http_server(self,
+                               node.public_address,
+                               host_port, b"/is_http")
+
+        def _get_datasets(unused_arg):
+            return cluster.client.list_datasets_configuration()
+        d.addCallback(_get_datasets)
+
+        def _verify_created_volume_is_size(datasets):
+            dataset = next(d for d in datasets
+                           if d.metadata.get(u'name') == volume_name)
+            self.assertEqual(int(dataset.metadata.get(u'maximum_size')),
+                             parse_num(size))
```

Comment:
I was suggesting starting with a number, which would be used here, and converting that to a size expression to pass to plugin via docker.

That is, instead of generating a string and then parsing it, start with a number and format it as a string.